File: Javascript/src/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const dictionary = {};
  let i = 0;
  for (const val of nums) {
    if (target - val in dictionary) return [dictionary[target - val], i];
    dictionary[val] = i++;

  }
  return [-1, -1];
};

nums = [2,7,11,15], target = 9
console.log(twoSum(nums,target));

File: Java/interview_preparation/src/main/java/com/cleo/interview_preparation/Two_Sum.java
package com.cleo.interview_preparation;

import java.util.HashMap;
import java.util.Map;

public class Two_Sum {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        int i=0;
        for(int num:nums){
            if(map.containsKey(target-num)){
                return new int[]{i,map.get(target-num)};

            }
            map.put(num, i);
            i++;
        }
        return new int[]{-1,-1};
        
    }
    
}
